### DISCUSSION TITLE
Strange binding issue

### DISCUSSION THREAD

**Discussion by maxigit** [qutebrowser/qutebrowser#8019] [General]
I've created a binding to  go to the first input and set username and password automatically. I have the two following bindings

- `<Ctrl+z>l fake-key -g gi<Ctrl-z>l` (normal mode)
- `<Ctrl+z>l :spawn --userscript qute-pass --unfiltered` (insert mode)

Basically `<Ctrl-z>l` does the same as `zl` but in insert mode and `<Ctrl-z>l` in normal mode, calls `gi` to go to the first input and then call `<Ctrl-z>`. It works but the problem is I get the  `gi` keys inserted in the username input field (and then user name).

---

*(edited by @The-Compiler to add code tags, so that your `<Keybinds>` don't get hidden as invalid HTML tags)*

**Comment by The-Compiler** [comment]
This seems odd indeed. I opened #8031 to track this.

However, your bindings seem unnecessarily complex. Why define a binding which fakes a keypress to trigger another binding? Just bind it to the proper command in the first place (as shown by `:bind gi`):

- `hint inputs --first ;; spawn --userscript qute-pass --unfiltered`

**Reply by maxigit** [reply]
Sure, I can "expand" the binding (it was just easier to simulate keypresses than remembering and typing the full command).
However this is a good workaround and it works, thanks :-)
